starport/interface/cli/starport/cmd/serve.go

```diff
@@ -23,17 +26,28 @@ func NewServe() *cobra.Command {
 	c.Flags().StringVarP(&appPath, "path", "p", "", "Path of the app")
 	c.Flags().BoolP("verbose", "v", false, "Verbose output")
 	c.Flags().BoolP(flagForceReset, "f", false, "Force reset of the app state on start and every source change")
-	c.Flags().BoolP(flagResetOnce, "r", false, "Reset of the app state on first start")
+	c.Flags().BoolP(flagResetOnce, "r", false, "Reset of -the app state on first start")
+	c.Flags().StringP(flagConfig, "c", "", "Starport config file (default: ./config.yml)")
 
 	return c
 }
 
 func serveHandler(cmd *cobra.Command, args []string) error {
-	// create the chain
 	chainOption := []chain.Option{
 		chain.LogLevel(logLevel(cmd)),
 		chain.KeyringBackend(chaincmd.KeyringBackendTest),
 	}
+
+	// check if custom config is defined
+	config, err := cmd.Flags().GetString(flagConfig)
+	if err != nil {
+		return err
+	}
+	if config != "" {
+		chainOption = append(chainOption, chain.ConfigFile(os.ExpandEnv(config)))
```

Comment:
We can I think, pass this option anyway even if config path is an empty string but check for an empty string inside `chain.ConfigFile` to fallback to `conf.FileNames`.

Reply:
Mmmh then we don't use the benefit of this pattern which is to have optional parameters?
Or else we use a positional argument